### PR TITLE
Environment and image build improvements

### PR DESCRIPTION
**Pull request by Den4ik**
This PR is replace for https://github.com/davidalger/warden/pull/357

It would be great have ability to define repositories via CI variables or at warden `.env` files.
This PR provide improvements for image build and environment usage:
* Allow define custom php source and push repository at image build process
* Allow define custom warden source repository for environments

**Ref. PR** https://github.com/davidalger/warden/pull/345

### CODE REVIEW

File: images/scripts/build.sh [hunk 1]
```diff
@@ -4,13 +4,13 @@ trap 'error "$(printf "Command \`%s\` at $BASH_SOURCE:$LINENO failed with exit c
...
-    )"
-  )" >/dev/null \
-  && pwd
+    cd "$(
+        dirname "$(
+            (readlink "${BASH_SOURCE[0]}" || echo "${BASH_SOURCE[0]}") |
+                sed -e "s#^../#$(dirname "$(dirname "${BASH_SOURCE[0]}")")/#"
+        )"
+    )" >/dev/null &&
+        pwd
```
Comment (ihor-sviziev) [inline images/scripts/build.sh:13]: Please revert code re-formatting in the file

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
@davidalger, could you review this PR?

**Comment by davidalger** [comment]
Merge commit 4daedec. Made a couple tweaks, to keep the docker.io prefix on the names, and use var name WARDEN_IMAGE_REPOSITORY vs WARDEN_SOURCE_REPOSITORY.

**Comment by davidalger** [comment]
Thanks for the efforts here! It's good flexibility to have.

**Pull request merged**